components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadMiddleware.kt

```diff
@@ -103,16 +104,17 @@ class DownloadMiddleware(
     private fun restoreDownloads(store: Store<BrowserState, BrowserAction>) = scope.launch {
         downloadStorage.getDownloads().collect { downloads ->
             downloads.forEach { download ->
-                if (!store.state.downloads.containsKey(download.id)) {
+                if (!store.state.downloads.containsKey(download.id) && !download.private) {
```

Comment:
If we never save them and never restore them... then we do not need the private flag in the database anymore? Should we drop it and as part of the database migration also drop all downloads with private = true from the database?

Reply:
👍

Reply:
I pushed it here https://github.com/mozilla-mobile/android-components/pull/8526/commits/edcb645aeb03ee57f8f7b6b714e9f216d0548f0e :)